gradle/publish.gradle

```diff
@@ -0,0 +1,78 @@
+apply plugin: 'com.novoda.build-properties'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: 'maven-publish'
+
+buildProperties {
+    cli {
+        using(project)
+    }
+
+    bintray {
+        def bintrayCredentials = {
+            return isDryRun() ?
+                    ['bintrayOrg': 'n/a', 'bintrayRepo': 'n/a', 'bintrayUser': 'n/a', 'bintrayKey': 'n/a'] :
+                    new File("${System.getenv('BINTRAY_PROPERTIES')}")
+        }
+        using(bintrayCredentials()).or(cli)
+        description = '''This should contain the following properties:
+                        - bintrayOrg: name of the Bintray organisation to deploy the artifacts to
+                        - bintrayRepo: name of the repo of the organisation to deploy the artifacts to
+                        - bintrayUser: name of the account used to deploy the artifacts
+                        - bintrayKey: API key of the account used to deploy the artifacts'''
+                .stripIndent()
+    }
+}
+
+
+def pomConfig = {
+    licenses {
+        license {
+            name "The Apache Software License, Version 2.0"
+            url "http://www.apache.org/licenses/LICENSE-2.0.txt"
+            distribution "repo"
+        }
+    }
+    developers {
+        developer {
+            id "novoda"
+            name "Novoda GmbH"
+        }
+    }
+    scm {
+        url "https://github.com/novoda/aws-v4-signer-java"
+    }
+}
+
+bintray {
```

Comment:
it seems like this might be missing some other properties when compared to [novoda/bintray-release](https://github.com/novoda/bintray-release/blob/master/plugin/core/src/main/groovy/com/novoda/gradle/release/BintrayConfiguration.groovy#L20-L42) implementation. I don't think any of those is strictly needed, but I wonder if without setting `dryRun` we would be able to skip the upload if we wanted to.

Reply:
I brought back `dryRun`, let's see if we need anything else.

Reply:
For some reason, `dryRun = isDryRun()` doesn't work and is always true where inlining the method works expected `dryRun = project.buildProperties.cli['dryRun'].or(true).boolean` 😕

Reply:
Btw. according to the logs it picks up everything needed 🎉 

> Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/libs/aws-v4-signer-ios-ios-0.0.1-sources.jar'.
Uploading to https://api.bintray.com/content/n/a/n/a/aws-v4-signer/0.0.1/com/novoda/aws-v4-signer-ios/0.0.1/aws-v4-signer-ios-0.0.1.klib...
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/iOS/module.json'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/iOS/pom-default.xml'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/libs/aws-v4-signer-jvm-0.0.1-sources.jar'.
Uploading to https://api.bintray.com/content/n/a/n/a/aws-v4-signer/0.0.1/com/novoda/aws-v4-signer-jvm/0.0.1/aws-v4-signer-jvm-0.0.1.jar...
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/jvm/module.json'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/jvm/pom-default.xml'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/kotlinMultiplatform/module.json'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/kotlinMultiplatform/pom-default.xml'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/libs/aws-v4-signer-metadata-0.0.1-sources.jar'.
Uploading to https://api.bintray.com/content/n/a/n/a/aws-v4-signer/0.0.1/com/novoda/aws-v4-signer-metadata/0.0.1/aws-v4-signer-metadata-0.0.1.jar...
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/metadata/module.json'.
Skipping upload for missing file '/Users/tobiasheine/Documents/repos/aws-v4-signer-java/library/build/publications/metadata/pom-default.xml'.